ISC - Restructure - Learn and Build (#1415)
iotaledger/iota-wiki: docs/learn/smart-contracts/smart-contract-execution.md

---
description: An in-depth exploration of smart contract execution in blockchain technology, detailing their structure, invocation, and associated fees.
image: /img/logo/WASP_logo_dark.png
keywords:
- Smart Contracts
- Blockchain Technology
- Validators
- User Requests
- Gas Fees
---

# Smart Contract Execution

## TL;DR;

> - [Introduction](#introduction): Smart contracts are self-executing contracts on blockchain networks, activated by validators in response to user requests.

Reviewer: I'm not sure I understand what is meant with "self-executing contracts". I'd prefer to just call them programs, ran by validators upon request.

Author: ```suggestion
- [Introduction](#introduction): Smart contracts are programs run by validators upon request by validators in response to user requests.
```

Reviewer 2: Why is this resolved?

Author: don't know, it seems the suggestion didn't go through but it still resolved...weird. Anyhow, updated

- [Structure of a Smart Contract](#structure-of-a-smart-contract): Comprising functions within a decentralized framework, smart contracts have a unique state and accounts, managed by validator node consensus.
- [View Entry Points](#view-entry-points): Read-only operations in smart contracts that do not modify data or incur execution fees.
- [Full Entry Points](#full-entry-points): Functions that can alter a contract's state and typically require execution fees.
- [Invoking a Smart Contract](#invoking-a-smart-contract): Activation of smart contracts through requests, which can be off-ledger or on-ledger, with associated gas fees for computational resources.

## Introduction

> Smart contracts are essentially self-executing contracts, run on blockchain networks. Network nodes, also known as 

Reviewer 2: Same change needed as before I guess

Author: ```suggestion
Smart contracts are self-executing contracts run on blockchain networks. Network nodes, also known as 
```

## TL;DR;

- [Introduction](#introduction): Smart contracts are programs run by [validators](chain-owners-validators-and-access-nodes.md#validator-nodes) upon requests.
- [Structure of a Smart Contract](#structure-of-a-smart-contract): Comprising functions within a decentralized framework, smart contracts have a unique state and accounts, managed by validator node consensus.
- [View Entry Points](#view-entry-points): Read-only operations in smart contracts that do not modify data or incur execution fees.
- [Full Entry Points](#full-entry-points): Functions that can alter a contract's state and typically require execution fees.
- [Invoking a Smart Contract](#invoking-a-smart-contract): Activation of smart contracts through requests, which can be off-ledger or on-ledger, with associated gas fees for computational resources.

## Introduction

> Smart contracts are essentially programs run by upon requests. Network nodes, also known as 

Reviewer: I guess you meant to say something like this?

```suggestion
Smart contracts are essentially programs run by a network of nodes upon requests. Network nodes, also known as 
```
